spec/requests/posts_controller_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe 'PostsController', type: :request do
+  let(:user) { create(:user) } # Create user from factory
+
+  it 'renders the index action' do
+    get user_posts_path(user_id: user.id)
+    expect(response).to have_http_status(:ok)
+    expect(response).to render_template(:index)
+  end
```

Comment:
- [ ] As required for the project, kindly ensure that Spec file include tests that for each action you include test for the correct placeholder text of the body
Example : expect(response.body).to include("Welcome to user index page")